packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -840,7 +852,7 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
                     cell: inputCellFactory(
                         queryInfo,
                         editorModel,
-                        allowBulkRemove || allowBulkUpdate,
+                        !hideCheckboxCol && (allowBulkRemove || allowBulkUpdate),
```

Comment:
this check comes up a couple of times. maybe factor to an "allowSelectionCol" helper?